packages/checkout/widgets-lib/src/widgets/sale/hooks/useCurrency.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { useState, useEffect } from 'react';
+import { Passport } from '@imtbl/passport';
+
+import { Environment } from '@imtbl/config';
+import { TokenFilterTypes } from '@imtbl/checkout-sdk';
+import { providers } from 'ethers';
+import { PRIMARY_SALES_API_BASE_URL } from '../utils/config';
+
+import {
+  ClientConfig,
+  ClientConfigCurrency,
+  ClientConfigCurrencyConversion,
+  SaleWidgetCurrency,
+  SaleWidgetCurrencyType,
+  SignPaymentTypes,
+} from '../types';
+import { useCheckoutClient } from './useCheckoutClient';
+import { sortCurrencies } from '../utils/sortCurrencies';
+
+type ClientConfigResponse = {
```

Comment:
@mimi-imtbl consider applying the renaming consistently across. although i think this hook better keep the prev name as is not specific to currency at alone